packages/wonder-blocks-announcer/src/index.ts

```diff
@@ -0,0 +1,334 @@
+// TODO: publish wonder-blocks-style package WB-1776
+// import {srOnly} from "../../wonder-blocks-style/src/styles/a11y";
+import type {
+    PolitenessLevel,
+    RegionDef,
+    RegionDictionary,
+    RegionFactory,
+} from "./announcer.types";
+
+const TIMEOUT_DELAY = 5000;
+let announcer: Announcer | null = null;
+
+export type SendMessageProps = {
+    message: string;
+    level?: PolitenessLevel;
+    removalDelay?: number;
+};
+
+/**
+ * Public API method to send screen reader messages.
+ * @param {string} message The message to send.
+ * @param {PolitenessLevel} level Polite or assertive announcements
+ * @param {number} removalDelay Optional duration to remove a message after sending. Defaults to 5000ms.
+ * @returns {string} IDREF for targeted live region element
+ */
+export function sendMessage({
+    message,
+    level = "polite", // TODO: decide whether to allow other roles, i.e. role=`timer`
+    removalDelay,
+}: SendMessageProps): string {
+    announcer = Announcer.getInstance();
+
+    if (typeof jest === "undefined") {
+        setTimeout(() => {
+            return announcer?.announce(message, level, removalDelay);
+        }, 100);
+    } else {
+        // If we are in a test environment, announce without waiting
+        return announcer.announce(message, level, removalDelay);
+    }
+    return "";
+}
+/**
+ * Public API method to clear screen reader messages after sending.
+ * Clears all regions by default.
+ * @param {string} id Optional id of live region element to clear.
+ */
+export function clearMessages(id?: string) {
+    if (id && document?.getElementById(id)) {
+        announcer?.clear(id);
+    } else if (document) {
+        announcer?.clear();
+    }
+}
+
+/**
+ * Internal class to manage screen reader announcements.
+ */
+class Announcer {
```

Comment:
**suggestion:** I'd recommend moving this class into a separate file as well. Sometimes we use an `internal/` folder for things like this, but up to you to decide what folder is best.

Side question: Do you think that `Announcer` can be tested independently? or is it better to rely on the exposed APIs for unit testing (the ones that you already created)?

Reply:
Good question about testing! It seems redundant to test the class itself, at least the parts that affect the DOM. But perhaps some unit testing for the singleton and some of the utility methods would be a good idea?

Reply:
totally agree! if you see that adding unit tests to the class add value, then go for it. I wouldn't want to add more tests just for the sake of increasing coverage 😄.